### DISCUSSION TITLE
Modal in UserMenuItem

### DISCUSSION THREAD

**Discussion by jamest-nz** [filamentphp/filament#5080] [Help] [answered]
Hello, I see that it's possible to [customise the User Menu](https://filamentphp.com/docs/2.x/admin/navigation#customizing-the-user-menu), including adding new items. But can one of these UserMenuItems open a modal on top of the current page, rather than redirecting to a route/url? When looking at the code for [Navigation\UserMenuItem](https://github.com/filamentphp/filament/blob/f4b7fdf91eac39d94325ebf6ed6e6f29fc1785b8/packages/admin/src/Navigation/UserMenuItem.php) I'm thinking not, but I'm not really any kind of Livewire expert.

Is such a thing possible? Or would having such an app-wide modal incur stupid overhead on each page load? Apologies if already answered, I searched and could not find much.

Thanks.

**Comment by danharrin** [comment, marked as answer]
Yeah I think that could be a big overhead. We already have a modal on every page, but since the user menu is outside the Livewire component, it is unable to open modals. So we'd need to introduce a new Livewire component on every page load to be able to open this modal.

**Comment by jamest-nz** [comment]
Appreciate the help, thanks
